rotkehlchen/chain/bitcoin/manager.py

```diff
@@ -183,66 +170,155 @@ def _query_blockchain_info_has_transactions(
-            try:
-                tx = BitcoinTx.deserialize_from_blockchain_info(entry)
-            except DeserializationError as e:
-                log.error(f'Failed to deserialize bitcoin transaction {entry} due to {e!s}')
-                continue
+        new_block_height = 0
+        for raw_tx_list in accounts_raw_tx_lists.values():
+            for entry in raw_tx_list:
+                try:
+                    tx = processing_fn(entry)
+                except (
+                    DeserializationError,
+                    KeyError,
+                    RemoteError,
+                    UnableToDecryptRemoteData,
+                ) as e:
+                    msg = f'Missing key {e!s}' if isinstance(e, KeyError) else str(e)
+                    log.error(f'Failed to process bitcoin transaction {entry} due to {msg}')
+                    continue
 
-            if tx.timestamp > to_timestamp:
-                continue  # Haven't reached the requested range yet. Skip tx.
-            if tx.timestamp <= latest_queried_ts:
-                break  # All new txs have been queried - break loop, save ts to cache, and return.
+                if tx.timestamp > to_timestamp:
+                    continue  # Haven't reached the requested range yet. Skip tx.
 
-            tx_list.append(tx)
+                if tx.block_height <= last_queried_block:
+                    break  # All new txs have been queried. Skip tx and return.
 
-        if len(tx_list) == 0:
-            log.debug(f'No new transactions found for bitcoin accounts {accounts_str}')
-            return []
+                tx_list.append(tx)
 
-        with self.database.conn.write_ctx() as write_cursor:
-            for account in accounts:
-                self.database.set_dynamic_cache(
-                    write_cursor=write_cursor,
-                    name=DBCacheDynamic.LAST_BITCOIN_TX_TS,
-                    value=tx_list[0].timestamp,  # first in the list is the latest tx
-                    address=account,
-                )
+            block_height = tx_list[0].block_height if len(tx_list) > 0 else last_queried_block
+            new_block_height = max(new_block_height, block_height)
+
+        return new_block_height, tx_list
+
+    def _query_blockchain_info_transactions(
+            self,
+            accounts: Sequence[BTCAddress],
+            options: dict[str, Any],
+    ) -> tuple[int, list[BitcoinTx]]:
+        """Query blockchain.info for transactions.
+
+        We use the single address /rawaddr endpoint rather than the multiaddr endpoint since
+        some types of TxIOs are omitted from the multiaddr endpoint.
+
+        Returns a tuple containing the latest queried block height and the list of txs.
+        """
+        results: dict[BTCAddress, list[dict[str, Any]]] = defaultdict(list)
+        for account in accounts:
+            base_url = f'{BLOCKCHAIN_INFO_BASE_URL}/rawaddr/{account}'
```

Comment:
as I mentioned in discord this is not good. Changing from multiaddr to rawaddr here can possibly lead to many many queries. Would it not make sense to figure out why the discrepancy and if not important enough still go with multiaddr? There must be a reason for it.